### ISSUE TITLE
Feature Request: Preview Number of Images Available per Satellite for Selected ROI

### ISSUE THREAD

**Issue by 2320sharon** [Doodleverse/CoastSeg#169]
I would like to propose a new feature that would greatly enhance the user experience and provide valuable information at a glance.

## Feature Description:

Allow users to preview the number of images available for each satellite they select. This feature should be implemented as a function that returns a set with the satellite name and the number of images available. This action should be allowed per Region of Interest (ROI).

## Proposed Flow:

User selects a Region of Interest (ROI) or multiple ROIs.
A widget or similar UI element displays the number of images available for each satellite for the selected ROI(s).
This feature would provide users with a quick overview of the data available for their selected region(s), helping them make more informed decisions about which satellites to use for their analysis.

I believe this feature would be a valuable addition to the platform. Please let me know if there are any questions or if further clarification is needed.

**Comment by dbuscombe-usgs** [comment]
Excellent idea. This would greatly help those who find downloading long time-series of imagery problematic, breaking the download into smaller more manageable chunks

**Comment by 2320sharon** [comment]
Using Canva I made a mockup of what the preview would look like
1. The user would hover over an ROI, then under ROI hover widget they would click go to see all the images available for that ROI ID
- If the user for some reason couldn't hover over an ROI they can enter an ROI ID and get the images available that way

## Initial State
![image](https://github.com/Doodleverse/CoastSeg/assets/61564689/0d777269-4fe9-477a-a40f-afb4657759b3)
## After Go Clicked
![image](https://github.com/Doodleverse/CoastSeg/assets/61564689/9005187b-2414-4ebd-8b6c-0adc720b50cc)
## What happens if the ROI ID they entered did not exist
![image](https://github.com/Doodleverse/CoastSeg/assets/61564689/57c73a8e-9d56-45df-b233-014502317f27)

**Comment by dbuscombe-usgs** [comment]
Amazing!

**Comment by 2320sharon** [comment]
Okay so it turns out the idea of having this widget on the map was rather ambitious. I thought it was a simple idea until I realized how many components would have to communicate to have this idea work. I was able to get a version working where when the user hovered over an ROI then clicked "get available images" it would show the images available, but the ROI's ID that was currently being hovered over wouldn't show. Additionally it takes a 2-3 seconds to the get the images available, so I can't just run this operation each time the user hovers over  a new ROI.

I know we have a lot of features we want to build into coastseg so I'm hesitant to spend too much time on this feature. @dbuscombe-usgs let me know what you think of this idea instead:

I think the best approach for now, would be to create a standalone button called 'preview available images', located near the download images button, shows a scrollable list that shows the images available for each ROI the user clicked. The output would look something like this.

```
ROI ID: abc123
L5 :  5 images
L7 :  5 images
L8 :  5 images
L9 :  5 images
S2 :  5 images

ROI ID: abc456
L5 :  7 images
L7 :  8 images
L8 :  5 images
L9 :  80 images
S2 :  5 images
```

**Comment by dbuscombe-usgs** [comment]
A preview listing the available imagery located next to the download button would, to my mind, be the perfect solution. The purpose of this is to give the user more info about whether or not they'd like to go ahead and push that download button, so your solution works perfectly fpr that.

I agree that there are other more pressing issues to work on, so I agree that this should only be implemented now if it involves minimal code refactoring

**Comment by 2320sharon** [comment]
I'll attempt to implement this idea. I believe I have almost all of the code necessary to create this widget. If I can't get it done by today I'll put my progress in a branch and move on with other tasks.

**Comment by 2320sharon** [comment]
![image](https://github.com/Doodleverse/CoastSeg/assets/61564689/9a784710-3409-4223-8b15-bf3d2acfcee0)
Alright I've got this working. I'll be release a development version of coastseg that includes this feature. This was FAR easier to implement and honestly I think its better.

**Comment by 2320sharon** [comment]
This feature has been implemented in `0.0.73dev3`
